### PR TITLE
fixes travis build matrix and tox to actually use python3

### PR DESCRIPTION
**Pull request by jofusa**
Ok I took a new path. As i was digging into tests that existed there were other issues then just py3 not being tested.

First is that while marathon versions were getting tested, the mesos version was fixed to 0.23.

Some changes:
- Adds a new docker fixture base class.
- New pytest fixture style docker containers to setup zk, mesos master/slave and marathon.
- marathon/mesos version are now pinned together in the .travis.yml
- Rewrote and removed the behave BDD style tests
- fixes request mocked tests for py3 (can't use fixture style decorator)

### CODE REVIEW

**Comment by jofusa** [comment]
Having two test runners in different directories really is making this a bear. Let me think on this. 

Also on a side note how attached to behave are you guys. The docker dependencies could be much easier to manage in py.test.

**Comment by solarkennedy** [comment]
Wow, this is quite an overhaul. My main concern is that we are losing significant test coverage with this. The tests finish in a suspiciously fast 2 minutes, but we are no longer *actually* calling behave?

I'm not *that* attached to behave as much as I'm attached to the existing itests we have built up to actually exercise this code.

**Comment by jofusa** [comment]
Yeah i am still working on replicating the existing itests, not there entirely. 

We should not merge until we are happy that the new integration tests provide the similar coverage. I hope that these new fixtures make writing these easier. We use this fixture pattern extensively internally and have found it useful. 

If you have any issue getting them to run locally please let me know and i'll be happy to help sort through issues. 

I am using this library to drive at lot of our internal deployment infrastructure. So I am highly motivated to expand testing into the later versions of marathon.

**Comment by jofusa** [comment]
@solarkennedy take a look at the new integration tests. I have inlined the old BDD declarations to give an idea of the mapping. Let me know if i missed any case.

**Comment by solarkennedy** [comment]
Gotcha. 

In the behave tests we had an `after_scenario` that would purge all apps and tasks to give us a fresh start on every test setup. Is there an equivalent here? Or will leftover apps pollute subsequent test functions?

BDD versus the py.test format, I'm down with either one I guess.

I did like the docker-compose method versus the docker-py method, I found it easier to develop against. (`docker compose up`) I guess it is all the same, just in python from instead of yaml form?

@Rob-Johnson can you weigh in on your opinion?

**Comment by Rob-Johnson** [comment]
er, I'm not too sure, but here's some thoughts:

| First is that while marathon versions were getting tested, the mesos version was fixed to 0.23.
Thanks for noticing this. You're right that this isn't correct.

| Adds a new docker fixture base class.
I'm not convinced by this. We're replicating a lot of the work that docker-compose does, I don't really see the benefit we get by hand rolling this?

| Rewrote and removed the behave BDD style tests
I'm not sure I'm *for* the current implementation - we seem to lose lots of the things that make writing the integration tests easier, and I don't really see what we get from using py.test rather than behave here.

I understand that you want this to be python3 compatible, but you've only mentioned request_mock as being the blocker and you seem to have been able to fix that.

It's not that I'm against your use of py.test or anything, I'm just not convinced what *extra* we're getting by rewriting everything?

**Comment by jofusa** [comment]
Thanks for taking a look. I agree the hand rolled was unneeded. I have replaced the docker py containers with a docker-compose.yml. It breaks each component into its own image making testing against marathon/mesos versions easier.

Perhaps it is just a matter of tastes but py.test tests are easier for for me to grok. Also pytest fixtures are incredibly powerful, and should be able to replicate most features available to the behave tests. If there is something you think is missing let me know and ill add it in.

There is also a benefit to only having one test runner. 

Give it a try and let me know what you think

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@pytest.fixture(scope='function')
+def client():
+    client = marathon.MarathonClient(get_marathon_connection_string())
+
+    def ensure_connection():
+        assert client.get_info()
+    retry(ensure_connection, exception_to_retry=(AssertionError, marathon.MarathonError))
+
+    for app in client.list_apps():
+        client.delete_app(app.id, force=True)
```
Comment (jofusa) [inline tests/integration/conftest.py]: @solarkennedy this fixture will wipe marathon clean before every test

**Comment by solarkennedy** [comment]
First, I really appreciate your effort here. But I'm going to close this PR.

It is trying to do too many things at once. Here is a short list:
* Travis matrix fixes
* Initial migration of behave => pytest
* Conversion to docker-compose
* Makefile tox fixes

It is too much. Please try smaller changes and they will be more likely to get merged.

I don't think there is enough consensus that a migration from behave => pytest is actually worth it, but the other things are.

**Pull request closed without merging**